diff/diff.go

```diff
@@ -365,7 +366,26 @@ type Do func(a Event) (crud.Arg, error)
 
 func (sc *Syncer) eventLoop(d Do) error {
 	for event := range sc.eventChan {
-		err := sc.handleEvent(d, event)
+		// Stop if program is terminated
+		select {
+		case <-sc.stopChan:
+			return nil
+		default:
+		}
+
+		// For various reasons, Kong can temporarily fail to process
+		// a valid request (e.g. when the database is under heavy load).
+		// We retry each request up to 3 times on failure, after around
+		// 1 second, 3 seconds, and 9 seconds (randomized exponential backoff).
+		exponentialBackoff := backoff.NewExponentialBackOff()
+		exponentialBackoff.InitialInterval = 1 * time.Second
+		exponentialBackoff.Multiplier = 3
+		backoffObject := backoff.WithMaxRetries(exponentialBackoff, 4)
```

Comment:
Please declare backoffObject as a global variable and initialize it in an `init()`. That keeps this function (or rather handleEvent fn after refactoring) cleaner. Readability and maintainability in this part of the code is more important than usual because it is easy to trip over.

Reply:
done